fix(compiler-cli): avoid fatal diagnostics for missing template files

clydin opened the pull request:
A build will still fail in this case. However, for the language service, this allows the component to exist in the compiler registry and prevents cascading diagnostics within an IDE due to "missing" components. The originating template related errors will still be reported in the IDE. This case is particularly important when a template file either does not exist or is inaccessible to the language service.

devversion commented on packages/compiler-cli/src/ngtsc/annotations/component/src/handler.ts:
```diff
@@ -662,50 +663,66 @@ export class ComponentDecoratorHandler
+        if (e instanceof FatalDiagnosticError) {
+          diagnostics ??= [];
+          diagnostics.push(e.toDiagnostic());
+          isPoisoned = true;
```
do you know if `isPoisoned` should also be set if there are actual non-fatal template "errors"?

clydin replied:
This PR leaves that behavior as-is.

From a more general perspective, I'm not sure how template syntax errors would effect TCB generation or if they are handled somewhere later.  Might be something to investigate as a potential optimization as a followup.

clydin committed:
fix(compiler-cli): avoid fatal diagnostics for missing template files

A build will still fail in this case. However, for the language service,
this allows the component to exist in the compiler registry and prevents
cascading diagnostics within an IDE due to "missing" components. The
originating template related errors will still be reported in the IDE.
This case is particularly important when a template file either does
not exist or is inaccessible to the language service.

AndrewKushnir commented:
This PR was merged into the repository by commit a1cacc5b17d6f865bb260f475f1d2ef37dc845f1.

The changes were merged into the following branches: main
